src/components/FlowRunsBarChart.vue

```diff
@@ -119,9 +121,12 @@
   }
 
   function organizeFlowRunsWithGaps(flowRuns: FlowRun[]): (FlowRun | null)[] {
-    const { expectedStartTimeAfter, expectedStartTimeBefore } = toValue(props.filter).flowRuns ?? {}
+    const flowRunsFilter = toValue(props.filter).flowRuns
+    const { expectedStartTimeAfter, expectedStartTimeBefore } = flowRunsFilter?.expectedStartTimeAfter && flowRunsFilter.expectedStartTimeBefore
+      ? flowRunsFilter
+      : getTimeRangeFromFlows(flowRuns)
```

Comment:
This is pretty hard to grok. I'd extract this into its own method called `getTimeRange` and do something like this

```javascript
const { expectedStartTimeAfter, expectedStartTimeBefore } = toValue(props.filter).flowRuns ?? {}

if(expectedStartTimeAfter && expectedStartTimeBefore) {
  return { expectedStartTimeAfter, expectedStartTimeBefore }
}

return getTimeRangeFromFlows(flowRuns)
```